plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```
@@ -0,0 +1,572 @@
+                            Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                    .map(IgniteTemplateType::valueOf)
+                                    .filter(template -> template != IgniteTemplateType.PARTITIONED)
+                                    .ifPresent(value -> properties.put(property, value));
+                            break;
+                        case IgniteTableProperties.WRITE_SYNCHRONIZATION_MODE_PROPERTY:
+                            Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                    .map(IgniteWriteSyncMode::valueOf)
+                                    .filter(mode -> mode != IgniteWriteSyncMode.FULL_SYNC)
+                                    .ifPresent(value -> properties.put(property, value));
+                            break;
+                        case IgniteTableProperties.CACHE_NAME_PROPERTY:
+                        case IgniteTableProperties.CACHE_GROUP_PROPERTY:
+                            Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                    .filter(name -> !name.equals("SQL_PUBLIC_" + tableName))
+                                    .ifPresent(value -> properties.put(property, value));
+                            break;
+                        case IgniteTableProperties.DATA_REGION_PROPERTY:
+                            Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH))).ifPresent(value -> properties.put(property, value));
+                            break;
+                        default:
+                            throw new IllegalStateException("Unexpected value: " + property);
+                    }
+                }
+                return properties.build();
+            }
+        }
+        catch (SQLException throwables) {
+            throwables.printStackTrace();
```

Please do not swallow exception, Please throw `TrinoException` here.